Raven.Database/Raft/Storage/Handlers/ReplicationStateCommandHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Rachis;
+using Raven.Abstractions.Extensions;
+using Raven.Bundles.Replication.Data;
+using Raven.Database.Raft.Commands;
+using Raven.Database.Server.Tenancy;
+
+namespace Raven.Database.Raft.Storage.Handlers
+{
+    public class ReplicationStateCommandHandler : CommandHandler<ReplicationStateCommand>
+    {
+        public ReplicationStateCommandHandler(DocumentDatabase database, DatabasesLandlord landlord,RaftEngine engine) : base(database, landlord)
+        {
+            MaxReplicationLatency = database.Configuration.Cluster.MaxReplicationLatency;
+            Engine = engine;
+        }
+
+        private RaftEngine Engine { get; set; }
+
+        //default should be 2 minutes
+        public TimeSpan MaxReplicationLatency { get; set; }
+
+        public override void Handle(ReplicationStateCommand command)
+        {
+            //If we are the leader we won't have documents sent from ourself so it is a noop.
+            if (Engine.State == RaftEngineState.Leader)
+                return;
+            var lastChanges = command.DatabaseToLastModified;
+            foreach (var databaseName in lastChanges.Keys)
+            {
+                Task<DocumentDatabase> databaseTask;
+                if(Landlord.TryGetOrCreateResourceStore(databaseName, out databaseTask) == false)
+                    throw new InvalidOperationException($"ReplicationStateCommand could not load database: {databaseName}");
+                var database = databaseTask.Result;
```

Comment:
We just check the last index in the log, not whatever it is committed

On Jul 20, 2016 11:47 PM, "talweiss1982" <notifications@github.com> wrote:

> In Raven.Database/Raft/Storage/Handlers/ReplicationStateCommandHandler.cs
> <https://github.com/ayende/ravendb/pull/4040#discussion_r71603541>:
>
> > +
> > +        //default should be 2 minutes
> > +        public TimeSpan MaxReplicationLatency { get; set; }
> > +
> > +        public override void Handle(ReplicationStateCommand command)
> > +        {
> > +            //If we are the leader we won't have documents sent from ourself so it is a noop.
> > +            if (Engine.State == RaftEngineState.Leader)
> > +                return;
> > +            var lastChanges = command.DatabaseToLastModified;
> > +            foreach (var databaseName in lastChanges.Keys)
> > +            {
> > +                Task<DocumentDatabase> databaseTask;
> > +                if(Landlord.TryGetOrCreateResourceStore(databaseName, out databaseTask) == false)
> > +                    throw new InvalidOperationException($"ReplicationStateCommand could not load database: {databaseName}");
> > +                var database = databaseTask.Result;
>
> don't we check the commit index for election?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ayende/ravendb/pull/4040/files/2aed8fd7a321c05ba657fde148dcad5e96d5cfc5..0c63b065447ed6f54a8eedc4d940349a7ee88eda#r71603541>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4wAEsXxvtKpHuQg5MJdecBRiEUVt1cks5qXolWgaJpZM4JQ6zO>
> .
>